pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnectionPool.java

```diff
@@ -19,40 +19,18 @@
 package org.apache.pulsar.proxy.server;
 
 import io.netty.channel.EventLoopGroup;
-import java.io.IOException;
-import java.util.concurrent.ExecutionException;
+import io.netty.resolver.dns.DnsNameResolver;
+import java.util.Optional;
 import java.util.function.Supplier;
 import org.apache.pulsar.client.api.PulsarClientException;
 import org.apache.pulsar.client.impl.ClientCnx;
 import org.apache.pulsar.client.impl.ConnectionPool;
 import org.apache.pulsar.client.impl.conf.ClientConfigurationData;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class ProxyConnectionPool extends ConnectionPool {
     public ProxyConnectionPool(ClientConfigurationData clientConfig, EventLoopGroup eventLoopGroup,
-            Supplier<ClientCnx> clientCnxSupplier) throws PulsarClientException {
-        super(clientConfig, eventLoopGroup, clientCnxSupplier);
+                               Supplier<ClientCnx> clientCnxSupplier, DnsNameResolver dnsNameResolver)
+            throws PulsarClientException {
+        super(clientConfig, eventLoopGroup, clientCnxSupplier, Optional.of(dnsNameResolver));
     }
-
-    @Override
-    public void close() throws IOException {
-        log.info("Closing ProxyConnectionPool.");
-        pool.forEach((address, clientCnxPool) -> {
-            if (clientCnxPool != null) {
-                clientCnxPool.forEach((identifier, clientCnx) -> {
-                    if (clientCnx != null && clientCnx.isDone()) {
-                        try {
-                            clientCnx.get().close();
-                        } catch (InterruptedException | ExecutionException e) {
-                            log.error("Unable to close get client connection future.", e);
-                        }
-                    }
```

Comment:
Interestingly, it looks like this logic might have resulted in a `clientCnx` leak, so removing this could be a positive side effect. It differs from the `super.close()` in that the `ConnectionPool` implementation calls `future.thenAccept(ClientCnx::close);` if the future is not done.

Reply:
Yes, it was odd that the logic was different than the logic in the ConnectionPool. The class was added in #1821 with this note:
> Since the ConnectionPool used in proxy uses the same event worker group as the Proxy service we can't close the event workgroup when we close the client, hence overriding the close() function in ProxyConnectionPool to just close the connections and not the event worker group.

Closing the event workgroup was removed from ConnectionPool.close in #13836. It should not have been closed in that method in the first place.

What I don't fully understand is that #1821 didn't copy the ConnectionPool.close logic completely.